libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/location/LocationEngineConductor.java

```diff
@@ -45,20 +45,35 @@ public LocationEngine obtainLocationEngine() {
     return locationEngine;
   }
 
-  private void initLocationEngine(Context context, boolean simulateRoute) {
+  private void initialize(Context context, LocationEngine locationEngine, boolean simulateRoute) {
     if (simulateRoute) {
-      locationEngine = new ReplayRouteLocationEngine();
+      this.locationEngine = new ReplayRouteLocationEngine();
```

Comment:
Have we considered the option to simulate including a custom `LocationEngine`? Right now, if `options.shouldSimulateRoute()` is `true` we're creating a `ReplayRouteLocationEngine` and ignoring the potential custom `LocationEngine` passed in. Shouldn't be the other way around? I.e. if there's a custom location engine in play we don't rely on the simulation option at all 🤔 Does that make sense? In any case, as both approaches are valid let's make sure we update the docs clarifying this.

Reply:
@Guardiola31337 yeah let's favor the custom engine before simulation if a custom engine is provided and simulation is enabled